R/compat.r

```diff
@@ -18,3 +18,25 @@ warn_unicode_on_windows <- function(code, warn) {
         if (real_len != r_len) warn(UNICODE_WARNING)
     }
 }
+
+
+DEVICE_ERR <-
+"Your R installation is misconfigured:
+getOption('bitmapType') == '%s', but capabilities('%s') == FALSE.
+This means that the png() device is unable to work.
+Please %s."
+
+preflight_check <- function() {
+    bitmap_type <- getOption('bitmapType')
+    xlib  <- capabilities('X11')
+    cairo <- capabilities('cairo')
+    
+    insane_xlib  <- bitmap_type == 'Xlib'  && !xlib
+    insane_cairo <- bitmap_type == 'cairo' && !cairo
+    if (insane_xlib || insane_cairo) {
+        help_msg <-
+            if (!xlib && !cairo) 'install an R that is configured to have either Xlib or cairo support'
+            else sprintf("put `options(bitmapType='%s')` into your ~/.Rprofile", if (xlib) 'Xlib' else 'cairo')
```

Comment:
if we can change the abort to a user-visible, non-suppressible warning without having a invasive change to our code, let’s do it! i also don’t want to make the check more than once, that wouldn’t make sense.

but honestly, if the sysadmin never once tested the thing (s)he installed, (s)he does a pretty bad job and deserves angry users.